src/Geno/C++/GUI/Widgets/FindInWorkspace.cpp

```diff
@@ -0,0 +1,172 @@
+ * applications, and to alter it and redistribute it freely, subject to the following restrictions:
+ *
+ * 1. The origin of this software must not be misrepresented; you must not claim that you wrote the
+ *    original software. If you use this software in a product, an acknowledgment in the product
+ *    documentation would be appreciated but is not required.
+ * 2. Altered source versions must be plainly marked as such, and must not be misrepresented as
+ *    being the original software.
+ * 3. This notice may not be removed or altered from any source distribution.
+ */
+
+#include "FindInWorkspace.h"
+#include "Application.h"
+#include "Components/Workspace.h"
+#include "GUI/PrimaryMonitor.h"
+#include "GUI/MainWindow.h"
+#include "TextEdit.h"
+#include "TitleBar.h"
+#include "Auxiliary/ImGuiAux.h"
+
+#include <filesystem>
+#include <iostream>
+#include <sstream>
+#include <regex>
+
+#if defined( __linux__ ) || defined ( __APPLE__ )
+#include <sys/types.h>
+#include <sys/stat.h>
+#endif // __linux__
+
+//////////////////////////////////////////////////////////////////////////
+
+FindInWorkspace::FindInWorkspace( void )
+{
+
+} // FindInWorkspace
+
+//////////////////////////////////////////////////////////////////////////
+
+FindInWorkspace::~FindInWorkspace( void )
+{
+
+} // ~FindInWorkspace
+
+//////////////////////////////////////////////////////////////////////////
+
+bool CheckExtension( const std::string& rExtension )
+{
+	return rExtension == ".cpp" || rExtension == ".hpp" || rExtension == ".cxx" || rExtension == ".h" || rExtension == ".c";
+
+} // CheckExtension
+
+//////////////////////////////////////////////////////////////////////////
+
+void FindInWorkspace::Show( bool* pOpen )
+{
+	if( pOpen == 0 /* false */ || Application::Instance().CurrentWorkspace() == nullptr )
+		return;
+
+	//////////////////////////////////////////////////////////////////////////
+
+	m_WorkspacePath = "";
+	m_WorkspacePath = Application::Instance().CurrentWorkspace()->m_Location.string();
+
+	ImGuiIO& rIo = ImGui::GetIO();
+
+	ImGuiWindowFlags Flags = ImGuiWindowFlags_NoSavedSettings;
+
+	ImGui::Begin( "Find Files in Workspace", pOpen, Flags );
+
+	ImGui::SetWindowSize( ImVec2( 900, 400 ), ImGuiCond_FirstUseEver );
+	ImGui::SetWindowPos( ImVec2( rIo.DisplaySize.x * 0.5f, rIo.DisplaySize.y * 0.5f ), ImGuiCond_FirstUseEver );
+
+	ImGuiTableFlags TableFlags = ImGuiTableFlags_Borders | ImGuiTableFlags_RowBg | ImGuiTableFlags_ScrollX | ImGuiTableFlags_NoBordersInBody;
+
+	if( ImGui::BeginTable( "##FileTable", 3, TableFlags, ImVec2( ImGui::GetWindowSize().x - 4, ImGui::GetWindowSize().y * 0.85f ) ) )
+	{
+		ImGui::TableSetupColumn( "File" );
+		ImGui::TableSetupColumn( "Path" );
+		ImGui::TableSetupColumn( "Last Modified" );
+		ImGui::TableHeadersRow();
+
+		int TableRow = 0;
+
+		for( auto& rEntry : std::filesystem::recursive_directory_iterator( m_WorkspacePath ) )
```

Comment:
I'm not sure that a directory iterator is what we want to use here. We might have the workspace in a following file structure:
```
root
|-- build/
|   |-- Workspace.gwks
|-- src/
    |-- main.cpp
```
In which case this will not find any text in main.cpp since it's outside the workspace directory.
Also, it's unnecessarily searching files that are unrelated to the workspace.